theano/tensor/opt.py

```diff
@@ -3253,6 +3253,50 @@ def local_flatten_lift(node):
 ##################
 
 
+@register_canonicalize
+@register_specialize
+@gof.local_optimizer([T.Reshape])
+def local_useless_reshape(node):
+    """ Reshape{ndim=1}(vector) -> vector
+
+    Remove Reshape when both the input and the output have a single
+    dimension.
+    If the shape of the input can't be inferred, an Assert is inserted to
+    ensure that the given new shape is the same as the original one.
+    """
+    if isinstance(node.op, T.Reshape):
+        x, shape = node.inputs
+        if node.op.ndim == 1 and x.ndim == 1:
+            out = x
+
+            new_shape_len = None
+            try:
+                new_shape_len = get_scalar_constant_value(shape.shape[0])
+            except NotScalarConstantError:
+                pass
+            if new_shape_len != 1:
+                out = assert_op(out, T.eq(shape.shape[0], 1))
```

Comment:
You mean 3269, the node.op.ndim check ?
It's to prevent errors like :
```
y = x.reshape(z, ndim=1)
f = function([x,z], y)
f([0,1,2,3], [2,2])
```